asmcomp/power/emit.mlp

```diff
@@ -416,15 +422,26 @@ module BR = Branch_relaxation.Make (struct
         else 5
     | Lop(Istackoffset _) -> 1
     | Lop(Iload {memory_chunk; addressing_mode; _ }) ->
+      let loadinstr =
+        match memory_chunk with
+        | Thirtytwo_signed -> "lwa"
+        | Word_int | Word_val -> "ld"
+        | _ -> "neither_lwa_nor_ld"
+      in
```

Comment:
This introduces duplication with the code in `load_store`, plus the weird value `"neither_lwa_nor_ld"` which is sort of a hack. (Using strings instead of an algebraic datatype feels hackish in the first place, but this is making it slightly worse.)

Could you instead define a helper function `load_instr_for_chunk` or something, by extracting the code of the `Istore` emitter, and reuse it here?

Reply:
Done.